cmake/app/boilerplate.cmake

```diff
@@ -247,7 +247,15 @@ else() # NOT FIRST_BOILERPLATE_EXECUTION
 
   # Have the child image select the same BOARD that was selected by
   # the parent.
+  # Unless parent was "ns" in which case we assume that the child images are
+  # all are secure, and should be build using the secure version of the board.
   set(BOARD ${CACHED_BOARD})
+  if (${BOARD} MATCHES  ".*ns")
+    string(LENGTH ${BOARD} len)
+    MATH(EXPR len "${len}-2")
+    string(SUBSTRING ${BOARD} 0 ${len} BOARD)
```

Comment:
If there is a cleaner way to do this I will be very happy :D

Reply:
I believe the cleanest solution would be to not have two boards.

But this is out-of-scope for this release.

Not sure what the best temporary solution would be ...

Reply:
Do you have a proposal for how to do the substring operation "remove two from end" without taking the length of the string first?

Reply:
```
if(${BOARD} STREQUAL nrf9160_pca10090ns)
set(BOARD nrf9160_pca10090)
endif()

```

If this code lives long enough to become a portability issue we deserve the pain of having to fix it.

Reply:
I agree, changed it to this now.